pkg/controller/controller.go

```diff
@@ -445,7 +445,7 @@ func (p *csiProvisioner) Provision(options controller.VolumeOptions) (*v1.Persis
 	rep := &csi.CreateVolumeResponse{}
 
 	// Resolve provision secret credentials.
-	provisionerSecretRef, err := getSecretReference(provisionerSecretParams, createVolumeRequestParameters, pvName, options.PVC)
+	provisionerSecretRef, err := getProvisionSecretReference(provisionerSecretParams, createVolumeRequestParameters, pvName, options.PVC)
```

Comment:
Would it be simpler to just pass in a new PVC object with only name and namespace set? Like in delete?

Reply:
True, the new function was due to my mix up of PVC.Annotations and SC.Parameters. Changed to use a new PVC object with name and namespace set.